src/SmartTransactionsController.ts

```diff
@@ -330,35 +397,54 @@ export default class SmartTransactionsController extends BaseController<
     });
   }
 
-  async updateSmartTransactions() {
+  async updateSmartTransactions({
+    networkClientId,
+  }: {
+    networkClientId?: NetworkClientId;
+  } = {}): Promise<void> {
     const { smartTransactions } = this.state.smartTransactionsState;
-    const { chainId } = this.config;
+    const chainId = this.getChainId({ networkClientId });
+    const smartTransactionsForChainId = smartTransactions?.[chainId];
 
-    const currentSmartTransactions = smartTransactions?.[chainId];
-
-    const transactionsToUpdate: string[] = currentSmartTransactions
+    const transactionsToUpdate: string[] = smartTransactionsForChainId
       .filter(isSmartTransactionPending)
       .map((smartTransaction) => smartTransaction.uuid);
 
     if (transactionsToUpdate.length > 0) {
-      this.fetchSmartTransactionsStatus(transactionsToUpdate);
+      this.fetchSmartTransactionsStatus(transactionsToUpdate, {
+        networkClientId,
+      });
     }
   }
 
-  async confirmSmartTransaction(smartTransaction: SmartTransaction) {
+  // TODO make this a private method?
+  async confirmSmartTransaction(
+    smartTransaction: SmartTransaction,
+    {
+      chainId,
+      ethQuery,
+    }: {
+      chainId: Hex;
+      ethQuery: EthQuery;
+    },
+  ) {
     const txHash = smartTransaction.statusMetadata?.minedHash;
     try {
-      const transactionReceipt =
-        await this.ethersProvider.getTransactionReceipt(txHash);
-      const transaction = await this.ethersProvider.getTransaction(txHash);
+      const transactionReceipt = await query(
```

Comment:
`query` returns `any`, so we should give it a type:

``` suggestion
      const transactionReceipt: { maxFeePerGas?: string; maxPriorityFeePerGas?: string; blockNumber: string } | null = await query(

Reply:
Done here: https://github.com/MetaMask/smart-transactions-controller/pull/210/commits/94e2fac1cfdfedacd31b038a72815abb013dc19c